molgenis-data-rest/src/main/java/org/molgenis/data/rest/v2/AttributeResponseV2.java

```diff
@@ -74,7 +73,10 @@ public AttributeResponseV2(final String entityParentName, EntityType entityType,
 			if (includeCategories && (this.fieldType.equals(AttributeType.CATEGORICAL) || this.fieldType.equals(
 					AttributeType.CATEGORICAL_MREF)))
 			{
-				this.categoricalOptions = dataService.findAll(refEntity.getId())
+				String refEntityIdAttributeName = refEntity.getIdAttribute().getName();
+				Query<Entity> refEntityQuery = new QueryImpl<>().pageSize(1000)
```

Comment:
What happens in case of more than a thousand categories?

Reply:
We could just add this to the api specs, and even set the pageSize to something like 100. > 1000 results is not really a sensible usecase fore the 'includerefs' flag.